app/templating/schema_context.py

```diff
@@ -16,20 +16,20 @@ def build_schema_context(metadata, aliases, answer_store, routing_path, group_in
     return {
         'exercise': _build_exercise(metadata),
         'respondent': _build_respondent(metadata),
-        'answers': _map_alias_to_answers(aliases, answer_store, routing_path),
+        'answers': _map_alias_to_answers(schema, aliases, answer_store, routing_path),
         'group_instance': group_instance,
     }
 
 
-def _map_alias_to_answers(aliases, answer_store, routing_path):
+def _map_alias_to_answers(schema, aliases, answer_store, routing_path):
     values = {}
     for alias, answer_info in aliases.items():
         matching_answers = []
 
-        answers = answer_store.filter(answer_id=answer_info['answer_id'], limit=True)
+        answers = answer_store.filter(answer_ids=[answer_info['answer_id']], limit=True)
 
         for answer in answers:
-            if _is_answer_in_routing_path(routing_path, answer):
+            if answer['answer_id'] in PathFinder.get_answer_ids_on_routing_path(schema, routing_path):
```

Comment:
`get_answer_ids_on_routing_path` could be called outside of both loops? Or the answer_ids on the routing path could be passed in to build_schema_context removing the dependency on both schema and PathFinder.